src/sentry/api/endpoints/organization_metrics.py

```diff
@@ -492,3 +511,67 @@ def get(self, request: Request, organization: Organization) -> Response:
                 standard_meta=True,
             ),
         )
+
+
+@region_silo_endpoint
+class OrganizationMetricsMetadataEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.TELEMETRY_EXPERIENCE
+
+    """
+    Get metadata of a metric for a given set of projects in a time interval.
+    The current metadata supported for metrics is:
+    - Code locations -> these are the code location in which the metric was emitted.
+    """
+
+    def _extract_metric_meta_types(self, request: Request) -> Sequence[MetricMetaType]:
+        meta_types = []
+
+        for meta_type in MetricMetaType:
+            if request.GET.get(meta_type.value) == "true":
+                meta_types.append(meta_type)
+
+        return meta_types
+
+    def _get_metric_code_locations(
+        self,
+        request: Request,
+        organization: Organization,
+        projects: Sequence[Project],
+        start: datetime,
+        end: datetime,
+    ) -> Sequence[MetricCodeLocations]:
+        return get_metric_code_locations(
+            metric_mris=[request.GET["metric"]],
+            start=start,
+            end=end,
+            organization=organization,
+            projects=projects,
+        )
+
+    def get(self, request: Request, organization) -> Response:
```

Comment:
Should the response be paginated? How big is it and what is the expected growth rate?

Reply:
The endpoint is still in progress, it's based on a MVP so the code is not definitive, I could mark it as experimental.

Reply:
Starting this week we're enforcing pagination on all list endpoints even the experimental ones. The reason is that historically people launched MVPs not paginated and then usage went up over time and causes bugs and timeouts. 
It's also easier to build the frontend paginated from the beginning than converting it to paginated later.

Reply:
I completely understand. We will definitely implement pagination but it's something we can't technically do right now, because the endpoint is backed by a unordered data store that doesn't allow any kind of offsets + limits when querying data.

Reply:
In addition, since you can pass multiple metrics and those are stored in different "buckets" in the data store, we would have to do a lot of magic to return the data paginated. I would prefer to defer the decision to paginate once we have a stable database that serves code locations. As of now, they are just ephemeral in-memory.

Just for your information, we have a maximum of 5 code locations per metric right now, but this might change.

Reply:
Thanks for the context. [API TSC is building tests](https://sentry.slack.com/archives/C02KZAK9A0Y/p1709233382613219) to enforce paginations on the PR 😛 so this was a good example for us.

Reply:
To further make the endpoint more predictable, I added further limitations so that I know the maximum response size in case of big requests: https://github.com/getsentry/sentry/pull/66046.